sdk/cosmos/azure-cosmos/src/test/java/com/azure/cosmos/CosmosBulkTest.java

```
@@ -0,0 +1,401 @@
+
+    @Factory(dataProvider = "clientBuildersWithDirectSession")
+    public CosmosBulkTest(CosmosClientBuilder clientBuilder) {
+        super(clientBuilder);
+    }
+
+    @BeforeClass(groups = {"simple"}, timeOut = SETUP_TIMEOUT)
+    public void before_CosmosBulkTest() {
+        assertThat(this.bulkClient).isNull();
+        this.bulkClient = getClientBuilder().buildAsyncClient();
+        bulkAsyncContainer = getSharedMultiPartitionCosmosContainer(this.bulkClient);
+    }
+
+    @AfterClass(groups = {"simple"}, timeOut = SHUTDOWN_TIMEOUT, alwaysRun = true)
+    public void afterClass() {
+        safeCloseAsync(this.bulkClient);
+    }
+
+    @Test(groups = {"simple"}, timeOut = TIMEOUT)
+    public void createItem_withBulk() {
+        int totalRequest = getTotalRequest();
+
+        Flux<CosmosItemOperation> cosmosItemOperationFlux = Flux.merge(
+            Flux.range(0, totalRequest).map(i -> {
+                String partitionKey = UUID.randomUUID().toString();
+                TestDoc testDoc = this.populateTestDoc(partitionKey);
+
+                return BulkProcessingUtil.getBulkCreateItemOperation(testDoc, new PartitionKey(partitionKey));
+            }),
+            Flux.range(0, totalRequest).map(i -> {
+                String partitionKey = UUID.randomUUID().toString();
+                EventDoc eventDoc = new EventDoc(UUID.randomUUID().toString(), 2, 4, "type1", partitionKey);
+
+                return BulkProcessingUtil.getBulkCreateItemOperation(eventDoc, new PartitionKey(partitionKey));
+            }));
+
+        BulkProcessingOptions<CosmosBulkTest> bulkProcessingOptions = new BulkProcessingOptions<>();
+        bulkProcessingOptions.setMaxMicroBatchSize(100);
+        bulkProcessingOptions.setMaxMicroBatchConcurrency(5);
+
+        Flux<CosmosBulkOperationRequestAndResponse<CosmosBulkTest>> responseFlux = bulkAsyncContainer
+            .processBulkOperations(cosmosItemOperationFlux, bulkProcessingOptions);
```

could you please also add tests for case when the input flux `cosmosItemOperationFlux` errors.

---

Have added one Mo and will make it better. Right now we are eating up the errors passed, and I cannot find another way. I wanted to have a response for it too, with the operation/response null and user passed exception in CosmosBulkOperationResponse. Is ignoring the error signal valid way?